frame/support/procedural/src/pallet/parse/error.rs

```diff
@@ -70,12 +70,18 @@ impl ErrorDef {
 			.variants
 			.iter()
 			.map(|variant| {
-				if !matches!(variant.fields, syn::Fields::Unit) {
-					let msg = "Invalid pallet::error, unexpected fields, must be `Unit`";
-					return Err(syn::Error::new(variant.fields.span(), msg))
+				match &variant.fields {
+					Fields::Unit => {},
+					Fields::Named(f) if f.named.len() < 2 => {},
+					Fields::Unnamed(u) if u.unnamed.len() < 2 => {},
+					_ => {
+						let msg = "Invalid pallet::error, unexpected fields, must be `Unit` or \
```

Comment:
This will not really prevent users from putting whatever they like as parameter.

Reply:
That's what I thought as well, but there isn't really a way to dive into the type and check its encoded size. There was an idea to create an integrity test here to check for the encoded size, but then there's also the question of what I should check for. Simply checking if the size is less than 3 doesn't quite satisfy the criteria for allowing arbitrary nested error types, so I'm left with just allowing everything here.

Reply:
maybe we can do this

we can create a trait:
```
/// Any implementation of this trait should be a enum type with unit variant or variant containing a single type which implements NestedPalletError
/// The trait should be usually derived with `NestedPalletError` derive macro.
trait NestedPalletError {}
```

And we create a derive macro which implement this trait and ensure the constraint:
So the code:
```
#[derive(NestedPalletError)]
enum Foo<Bar> { A(Bar) }
```

will check that foo enum has only unit or single field variant and generate:
```
enum Foo { A(Bar) }
impl<Bar: NestedPalletError> NestedPalletError for Foo<Bar> {}
```

PS: NestedPalletError is just the first name that came to my mind, we can find better maybe

Reply:
Huh... adding additional configurations to foreign fields sounds a bit dirty to me, but I don't seem to see anything better...

Reply:
You already require that the foreign type implements `TypeInfo` trait :P